applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp

```diff
@@ -61,36 +65,45 @@ void TotalStructuralMassProcess::Execute()
         // We get the values from the condition
         const Properties& this_properties = it_elem->GetProperties();
 
+        element_mass = 0.0; // resetting the element-mass
+
         if (local_space_dimension == 0) { // POINT MASSES
             if (it_elem->Has(NODAL_MASS))
-                total_mass += it_elem->GetValue(NODAL_MASS);
+                element_mass = it_elem->GetValue(NODAL_MASS);
         } else if (local_space_dimension == 1) { // BEAM CASE
             const double density = GetFromProperty(this_properties,DENSITY);
             const double area = GetFromProperty(this_properties,CROSS_AREA);
-            total_mass += density * area * r_this_geometry.Length();
+            element_mass = density * area * r_this_geometry.Length();
         } else if (local_space_dimension == 2 && dimension == 3) { // SHELL-MEMBRANE
             const double area = r_this_geometry.Area();
             if (this_properties.Has(SHELL_ORTHOTROPIC_LAYERS)) { // composite material
                 const auto orthotropic_layers = this_properties[SHELL_ORTHOTROPIC_LAYERS];
                 for (std::size_t i=0; i<orthotropic_layers.size1(); ++i)
-                    total_mass += orthotropic_layers(i,0) * orthotropic_layers(i,2) * area; // thickness*density*area
+                    element_mass += orthotropic_layers(i,0) * orthotropic_layers(i,2) * area; // thickness*density*area
             } else {
                 const double thickness = GetFromProperty(this_properties,THICKNESS);
                 const double density = GetFromProperty(this_properties,DENSITY);
-                total_mass += density * thickness * area;
+                element_mass = density * thickness * area;
             }
         } else { // SOLID
             const double thickness = (dimension == 2) ? (this_properties.Has(THICKNESS)) ? this_properties[THICKNESS] : 1.0 : 1.0;
             const double volume = (dimension == 2) ? r_this_geometry.Area() : r_this_geometry.Volume();
             const double density = GetFromProperty(this_properties,DENSITY);
-            total_mass += density * thickness * volume;
+            element_mass = density * thickness * volume;
         }
 
-        // We restore the current configuration
-        for (std::size_t i_node = 0; i_node < number_of_nodes; ++i_node)
+        // We restore the current configuration and compute the nodal mass
+        for (std::size_t i_node = 0; i_node < number_of_nodes; ++i_node) {
             noalias(r_this_geometry[i_node].Coordinates()) = current_coordinates[i_node];
+            r_this_geometry[i_node].GetValue(NODAL_MASS) += element_mass / number_of_nodes;
```

Comment:
Be careful here, if the node is not locked, several elements might try to access the same node at the same time from different OpenMP threads.

Reply:
as discussed, I removed the omp now entirely bcs it is too dangerous with the coordinates-manipulations